dbms/src/TableFunctions/TableFunctionRandom.cpp

```diff
@@ -0,0 +1,477 @@
+        }
+        case TypeIndex::Set:
+            throw Exception("Type 'Set' can not be stored in a table.", ErrorCodes::LOGICAL_ERROR);
+        case TypeIndex::Interval:
+            throw Exception("Type 'Interval' can not be stored in a table.", ErrorCodes::LOGICAL_ERROR);
+        case TypeIndex::Nullable:
+            {
+                auto & column_nullable = typeid_cast<ColumnNullable &>(column);
+                auto nested_type = typeid_cast<const DataTypeNullable *>(type.get())->getNestedType();
+
+                auto & null_map = column_nullable.getNullMapData();
+                IColumn & nested_column = column_nullable.getNestedColumn();
+
+                fillColumnWithRandomData(nested_column, nested_type, limit, max_array_length, max_string_length, random_seed);
+
+                pcg32 generator(random_seed);
+                null_map.resize(limit);
+                for (UInt64 i = 0; i < limit; ++i)
+                {
+                    null_map[i] = generator() < 1024;
+                }
+                break;
+            }
+        case TypeIndex::Function:
+            throw Exception("Type 'Funclion' can not be stored in a table.", ErrorCodes::LOGICAL_ERROR);
+        case TypeIndex::AggregateFunction:
+            throw Exception("Random Generator not implemented for type 'AggregateFunction'.", ErrorCodes::NOT_IMPLEMENTED);
+        case TypeIndex::LowCardinality:
+            throw Exception("Random Generator not implemented for type 'LowCardinality'.", ErrorCodes::NOT_IMPLEMENTED);
+    }
+}
+
+StoragePtr TableFunctionRandom::executeImpl(const ASTPtr & ast_function, const Context & context, const std::string & table_name) const
+{
+    ASTs & args_func = ast_function->children;
+
+    if (args_func.size() != 1)
+        throw Exception("Table function '" + getName() + "' must have arguments.", ErrorCodes::LOGICAL_ERROR);
+
+    ASTs & args = args_func.at(0)->children;
+
+    if (args.size() < 2)
+        throw Exception("Table function '" + getName() + "' requires at least two arguments: "\
+                        " structure, limit(, max_array_length, max_string_length, random_seed).",
+                        ErrorCodes::NUMBER_OF_ARGUMENTS_DOESNT_MATCH);
+
+    if (args.size() > 5)
+        throw Exception("Table function '" + getName() + "' requires at most five arguments: "\
+                        " structure, limit, max_array_length, max_string_length, random_seed.",
+                        ErrorCodes::NUMBER_OF_ARGUMENTS_DOESNT_MATCH);
+
+    /// Parsing first argument as table structure and creating a sample block
+    std::string structure = args[0]->as<ASTLiteral &>().value.safeGet<String>();
+
+    UInt64 limit = 1;
+    UInt64 max_array_length = 10;
+    UInt64 max_string_length = 10;
+    UInt64 random_seed = 0; // zero for random
+
+    /// Parsing second argument if present
+    if (args.size() >= 2)
+        limit = args[1]->as<ASTLiteral &>().value.safeGet<UInt64>();
+
+    if (!limit)
```

Comment:
It's totally Ok, see `SELECT count() FROM numbers(0)`

Reply:
rewritten to table engine - outdated. No more limit in parameters